packages/code-snippet/src/CodeSnippetWidget.tsx

```diff
@@ -146,14 +155,17 @@ class CodeSnippetDisplay extends MetadataDisplay<
         notebookCell instanceof MarkdownCell &&
         snippet.metadata.language.toLowerCase() !== 'markdown'
       ) {
-        notebookCellEditor.replaceSelection(
+        notebookCellEditor.replaceSelection?.(
           this.addMarkdownCodeBlock(snippet.metadata.language, snippetStr)
         );
       } else {
-        notebookCellEditor.replaceSelection(snippetStr);
+        notebookCellEditor.replaceSelection?.(snippetStr);
+      }
+      const cell = notebookWidget.model?.contentFactory.createCodeCell({});
```

Comment:
does moving this into the if statement not cause issues with the usage outside the if statement below?